src/Symfony/Component/Filesystem/Watcher/FileChangeWatcher.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Filesystem\Watcher;
+
+use Symfony\Component\Filesystem\Watcher\Resource\Locator\FileResourceLocator;
+
+/**
+ * @internal
+ */
+class FileChangeWatcher implements WatcherInterface
+{
+    private $locator;
+
+    public function __construct()
+    {
+        $this->locator = new FileResourceLocator();
+    }
+
+    public function watch($path, callable $callback)
+    {
+        $resource = $this->locator->locate($path);
+
+        if (!$resource) {
+            throw new \InvalidArgumentException(sprintf('%s is not a valid path to watch', \gettype($path)));
+        }
+
+        $run = true;
+
+        while ($run) {
+            if ($changes = $resource->detectChanges()) {
+                foreach ($changes as $change) {
+                    $run = false !== $callback($change->getFile(), $change->getEvent());
+                }
+            }
+
+            sleep(1);
```

Comment:
Could be useful to use usleep and define by microseconds by an optional parameter?

Reply:
I'm not sure if it will be worth it, as the sleep is only applicable in this watcher, so I don't think it should be defined in every watcher as well as the interface.
And if we use `usleep` and the parameter is passed with a very low number, then it can be quite resource intensive to continuously check the filesystem and not have a long enough wait time between checks

Reply:
A parameter has been added to control the waiting time